### PR TITLE
Interpret basement data

### PR DESCRIPTION
**Pull request by buckley-w-david**
This PR is an initial run through of interpreting basement data.

PR is unfortunately larger than I would like, but it is one atomic addition isolated to new files.

https://trello.com/c/MHCxw57a/107-extract-load-foundation-data

### CODE REVIEW

**Comment by buckley-w-david** [comment]
[Extract/Load Foundation data](https://trello.com/c/MHCxw57a/107-extract-load-foundation-data)

File: python/src/energuide/embedded/basement.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            width = typing.cast(distance.Distance, floor.width)
+            floor_perimeter = (2 * length.metres) + (2 * width.metres)
+        else:
+            perimeter = typing.cast(distance.Distance, floor.perimeter)
+            floor_perimeter = typing.cast(float, perimeter.metres)
+
+        return Basement(
+            label=basement.get_text('Label'),
+            configuration_type=basement.xpath('Configuration/@type')[0],
+
```
Comment (sastels) [inline python/src/energuide/embedded/basement.py]: 🔥 blank line

File: python/tests/embedded/test_basement.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                <Section rank="1" percentage="100" rsi="0" nominalRsi="0" />
+            </Composite>
+        </ExteriorAddedInsulation>
+        <PonyWallType nominalInsulation="0">
+          <Description>User specified</Description>
+          <Composite>
+              <Section rank="1" percentage="100" rsi="0" nominalRsi="0" />
+          </Composite>
+        </PonyWallType>
+
```
Comment (sastels) [inline python/tests/embedded/test_basement.py]: blank line

**Comment by sastels** [comment]
minor comments, LGTM

File: python/src/energuide/embedded/basement.py [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+            'perimeterMetres': self.perimeter.metres if self.perimeter is not None else None,
+            'perimeterFeet': self.perimeter.feet if self.perimeter is not None else None,
+            'widthMetres': self.width.metres if self.width is not None else None,
+            'widthFeet': self.width.feet if self.width is not None else None,
+            'lengthMetres': self.length.metres if self.length is not None else None,
+            'lengthFeet': self.length.feet if self.length is not None else None,
+        }
+
+
+class BasementWall(_BasementWall):
```
[the remark this answers is not in the document]
Comment (buckley-w-david) [reply python/src/energuide/embedded/basement.py]: I could, but I think I'd have to override all it's methods, which would make the subclass pretty pointless.

File: python/src/energuide/embedded/basement.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    @property
+    def _header_area(self) -> area.Area:
+        return area.Area(self.perimeter.metres * self.height.metres)
+
+    def to_dict(self) -> typing.Dict[str, typing.Any]:
+        return {
+            'nominalRsi': self.nominal_insulation.rsi,
+            'nominalR': self.nominal_insulation.r_value,
+            'effectiveRsi': self.effective_insulation.rsi,
+            'effectiveR': self.effective_insulation.r_value,
```
[the remark this answers is not in the document]
Comment (buckley-w-david) [reply python/src/energuide/embedded/basement.py]: Yes, this PR was from before that refactor, I'll update it to take that into account

**Pull request merged**